### PR TITLE
Forbid votes while being muted

### LINKED ISSUE #3926: Disallow spec/kick/map-votes if person is muted
Spec/kick/map-vote reason field is still being abused

### PR DESCRIPTION
**Pull request by heinrich5991**
Fixes the problem of people abusing the vote reason for spamming.

Fixes #3926.

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/server/gamecontext.cpp [hunk 1]
```diff
@@ -4138,6 +4138,14 @@ bool CGameContext::RateLimitPlayerVote(int ClientID)
 	for(int i = 0; i < m_NumVoteMutes && !VoteMuted; i++)
 		if(!net_addr_comp_noport(&Addr, &m_aVoteMutes[i].m_Addr))
 			VoteMuted = (m_aVoteMutes[i].m_Expire - Server()->Tick()) / Server()->TickSpeed();
+	if(VoteMuted == 0)
```
Comment (def-) [inline src/game/server/gamecontext.cpp:4141]: I would remove the if

**Comment by murpii** [comment]
thank you heinrich

File: src/game/server/gamecontext.cpp (hunk 1 above)
Comment (heinrich5991) [reply src/game/server/gamecontext.cpp]: Done.

**Pull request merged**